src/jit/lowerxarch.cpp

```diff
@@ -925,7 +925,8 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
     // the below logic leads to conservative register specification.  In future when or if we find
     // one such case, this logic needs to be fine tuned for that case(s).
     if (varTypeIsByte(tree) || ((tree->OperGet() == GT_CAST) && varTypeIsByte(tree->CastToType())) ||
-        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) && varTypeIsByte(tree->gtGetOp2())))
+        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) &&
+         (varTypeIsByte(tree->gtGetOp2()) || tree->gtGetOp2()->IsCnsIntOrI())))
```

Comment:
IMO, Long term it is best that individual GenTree nodes that deal with byte types have logic to specify byteable reg specification in the corresponding TreeNodeInfoInit*() routine.
That way we can tailor the conditions for byteable register specification for each GenTree node and other GenTree nodes that don't deal with byte types don't need to be concerned about this logic.

Since you are modifying the condition for compare, how about taking this opportunity to move compare specific logic to TreeNodeInfoInitCmp()?

If you don't want to club that restructuring with this PR, please file a git issue to track it.